packages/frontend/src/components/DataViz/config/CartesianChartFieldConfiguration.tsx

```diff
@@ -154,26 +156,38 @@ const GroupByFieldAxisConfig = ({
     sqlColumns,
 }: {
     sqlColumns: VizSqlColumn[];
-
     field: undefined | { reference: string };
     groupByOptions?: VizPivotLayoutOptions[];
     actions: CartesianChartActionsType;
 }) => {
     const dispatch = useVizDispatch();
+    const error =
+        field !== undefined &&
+        !groupByOptions.find((x) => x.reference === field.reference)
+            ? `Column "${field.reference}" does not exist. Choose another`
+            : undefined;
     return (
         <FieldReferenceSelect
+            rightSection={
+                // When the field is deleted, the error state prevents the clear button from showing
+                error && (
```

Comment:
this a much better solution @rephus 

<img width="375" alt="image" src="https://github.com/user-attachments/assets/701af805-7017-4bc1-b428-4e198b182eb2">

no more onclick issues.